src/main/cpp/include/genomicsdb/genomicsdb_columnar_field.h

```diff
@@ -0,0 +1,326 @@
+/**
+ * The MIT License (MIT)
+ * Copyright (c) 2016 Intel Corporation
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of 
+ * this software and associated documentation files (the "Software"), to deal in 
+ * the Software without restriction, including without limitation the rights to 
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of 
+ * the Software, and to permit persons to whom the Software is furnished to do so, 
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all 
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS 
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR 
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER 
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN 
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+#ifndef GENOMICSDB_COLUMNAR_FIELD_H
+#define GENOMICSDB_COLUMNAR_FIELD_H
+
+#include "headers.h"
+#include "vcf.h"
+#include "gt_common.h"
+
+class GenomicsDBColumnarFieldException : public std::exception {
+  public:
+    GenomicsDBColumnarFieldException(const std::string m) : msg_(m) { ; }
+    ~GenomicsDBColumnarFieldException() { ; }
+    // ACCESSORS
+    /** Returns the exception message. */
+    const char* what() const noexcept { return msg_.c_str(); }
+  private:
+    std::string msg_;
+};
+
+/*
+ * Think of this class as a vector which cannot be resized
+ * The buffer space of this object is owned by GenomicsDBColumnarField - this object
+ * only contains offsets within the buffer
+ */
+template<class DataType>
+class LinkedVector
+{
+
+};
+//ceil(buffer_size/field_size)*field_size
+#define GET_ALIGNED_BUFFER_SIZE(buffer_size, field_size) ((((buffer_size)+(field_size)-1u)/(field_size))*(field_size))
+
+class GenomicsDBBuffer
+{
+  public:
+    GenomicsDBBuffer(const size_t num_bytes)
+    {
+      m_is_in_live_list = false;
+      m_buffer.resize(num_bytes);
+      m_valid.resize(num_bytes);
+      m_num_live_entries = 0ull;
+      m_num_filled_entries = 0ull;
+      m_num_unprocessed_entries = 0ull;
+      m_next_buffer = 0;
+      m_previous_buffer = 0;
+    }
+    //Delete both move and copy constructors
+    GenomicsDBBuffer(const GenomicsDBBuffer& other) = delete;
+    GenomicsDBBuffer(GenomicsDBBuffer&& other) = delete;
+    //Functions
+    inline void set_is_in_live_list(const bool val) { m_is_in_live_list = val; }
+    inline bool is_in_live_list() const { return m_is_in_live_list; }
```

Comment:
The convention I follow: if it's a boolean variable, the "get" function is name is_flag_set(). For other variables, get/set.